### PR TITLE
add confirmation option to actions

### PR DESCRIPTION
**Pull request by iantrich**
extracted from https://github.com/home-assistant/home-assistant-polymer/pull/3994

Docs: https://github.com/home-assistant/home-assistant.io/pull/10834

### CODE REVIEW

File: src/panels/lovelace/common/handle-click.ts [hunk 1]
```diff
@@ -30,6 +30,23 @@ export const handleClick = (
...
+    actionConfig.confirmation &&
+    (!actionConfig.confirmation.exemptions ||
+      !actionConfig.confirmation.exemptions.some(
+        (e) => e.user === hass!.user!.id
+      ))
+  ) {
+    if (
+      !confirm(
+        actionConfig.confirmation.text ||
+          "Are you sure you want to " + actionConfig.action + "?"
```
Comment (bramkragten) [inline src/panels/lovelace/common/handle-click.ts]: Use template literals.
```suggestion
          `Are you sure you want to ${actionConfig.action}?`
```

File: src/data/lovelace.ts [hunk 2]
```diff
@@ -45,24 +45,37 @@ export interface CallServiceActionConfig {
...
-export interface NoActionConfig {
+export interface NoActionConfig extends BaseActionConfig {
   action: "none";
 }
 
+export interface BaseActionConfig {
+  confirmation?: ConfirmationRestrictionTypeConfig;
+}
+
+export interface ConfirmationRestrictionTypeConfig {
```
Comment (bramkragten) [inline src/data/lovelace.ts]: Why the Type?
```suggestion
export interface ConfirmationRestrictionConfig {
```

File: src/data/lovelace.ts [hunk 3]
```diff
@@ -45,24 +45,37 @@ export interface CallServiceActionConfig {
...
+  text?: string;
+  exemptions?: RestrictionTypeConfig[];
+}
+
+export interface RestrictionTypeConfig {
```
Comment (bramkragten) [inline src/data/lovelace.ts]:
```suggestion
export interface RestrictionConfig {
```

**Comment by iantrich** [comment]
@bramkragten waiting for docs for double tap to merge first: https://github.com/home-assistant/home-assistant.io/pull/10808

**Comment by iantrich** [comment]
If you're good with this, we can merge and I'll get docs done after the double tap's are merged

**Review by bramkragten — APPROVED** [review]

**Pull request merged**